### DISCUSSION TITLE
Examples of the usage of the ontology and software implementation

### DISCUSSION THREAD

**Discussion by vChavezB** [OpenMath/om-rdf#7] [General]
Thanks for the serialization of OpenMath, I find this project interesting and I am trying to find how to use it.

My application consists of describing how to interpret formulas described in sensor specifications. To be specific field devices have parameters which can be obtained at runtime. The user needs to get the data from these parameters to calculate scaling, offset or a vendor specific calculation. The typical workflow is that the software developer reads the spec and then manually programs the formulas (PLC, microcontroller,etc). 

With OpenMath I plan to describe these formulas, e.g., if you multiply parameter 1 and parameter 2 from sensor A you can get temperature in celsius degrees (using qudt unit ontology for that).  

As far as I understand to declare these formulas you can use `openmath-cds.ttl` vocabulary and assert a `math:Application`. Then You need to assert Object properties of type `math:argument` with owl:range of type `rdf:list` or subclasses such as  `math:VariableList`.  Then you declare `rdf:first` and `rdf:rest`.

1. Are there any vocabularies or example ontologies where I can see how this is implemented in practice?
2. Are there any SPARQL  or SWRL examples where you interpret the `math:applications`. I found the project https://github.com/numerateweb/numerateweb but it would nice if I could get pointed where exactly could I look in the source code :) 

Thanks for your time.

**Comment by kenwenzel** [comment]
The rules can be also specified using POPCORN-LD:
https://github.com/numerateweb/numerateweb-rdf4j/blob/main/examples/src/main/resources/process-planning/planning.nwrules

These rules can then be directly translated to their RDF representation like in this example:
https://github.com/kenwenzel/numerateweb-swj-2022/blob/main/launch-ide/workspace/ProcessChains/imports/rules.ttl

Vocabulary from openmath-cds.ttl is not really required. This file is just an RDF translation of the existing OpenMath CDs to RDF. But it can serve as a source for example formulas.

If you are looking for examples on how to use the numerateweb Java Implementation you could take a look at the following project:
https://github.com/numerateweb/numerateweb-rdf4j
It is an OpenMath-RDF-based reasoner that executes mathematical computations.

**Reply by vChavezB** [reply]
Thanks a lot for the information. I will take a dive in the project and the links you provided 👍
